src/main/webapp/app/services/visualizations/visualization-grpc.directive.js

```diff
@@ -227,11 +229,26 @@
                 height = el.height();
             var panel = $('.grid-stack');
 
-            widgets[vm.widget].build(
-                visualMetadata,
-                el,
-                panel
-            );
+            if (!vm.isSaved) {
```

Comment:
no need to use ! . it could have been written like this .. if (vm.isSaved){....}else{....}